cleanlab/rank.py

```diff
@@ -28,76 +28,86 @@
 
 
 import numpy as np
+from typing import List
+import warnings
+from cleanlab.utils.label_quality_utils import subtract_confident_thresholds, get_entropy
 
 
 def order_label_issues(
-    label_issues_mask,
-    labels,
-    pred_probs,
+    label_issues_mask: np.array,
+    labels: np.array,
+    pred_probs: np.array,
     *,
-    rank_by="normalized_margin",
-):
-    """Sorts label issues by normalized margin.
+    rank_by: str = "normalized_margin",
+    rank_by_kwargs: dict = {},
+) -> np.array:
+    """Sorts label issues by label quality score.
+
+    Default label quality score is "normalized margin".
     See https://arxiv.org/pdf/1810.05369.pdf (eqn 2.2)
-    e.g. normalized_margin = prob_label - max_prob_not_label
 
     Parameters
     ----------
     label_issues_mask : np.array (bool)
       Contains True if the index of labels is an error, o.w. false
 
     labels : np.array
-      A discrete vector of noisy labels, i.e. some labels may be erroneous.
-      *Format requirements*: for dataset with K classes, labels must be in {0,1,...,K-1}.
+      Labels in the same format expected by the `score_label_quality()` method.
 
     pred_probs : np.array (shape (N, K))
-      P(label=k|x) is a matrix with K model-predicted probabilities.
-      Each row of this matrix corresponds to an example x and contains the model-predicted
-      probabilities that x belongs to each possible class.
-      The columns must be ordered such that these probabilities correspond to class 0,1,2,...
-      `pred_probs` should have been computed using 3 (or higher) fold cross-validation.
+      Predicted-probabilities in the same format expected by the `score_label_quality()` method.
 
-    rank_by : str ['normalized_margin', 'self_confidence']
+    rank_by : str ['normalized_margin', 'self_confidence', 'confidence_weighted_entropy']
       Method to order label error indices (instead of a bool mask), either:
         'normalized_margin' := normalized margin (p(label = k) - max(p(label != k)))
         'self_confidence' := [pred_probs[i][labels[i]] for i in label_issues_idx]
```

Comment:
[not your responsibility but could be quickly addressed before merging this PR] This can be more clearly written as:
```
Score by which to order label error indices (in increasing order), either: 'normalized_margin', 'self_confidence', or 'confidence_weighted_entropy'. See `score_label_quality()` documentation for description of these scores.

Reply:
And then I'd move the mathematical defn of the scores to the `score_label_quality()` docstring for its `method` arg. I'd also rewrite it a bit more clearly as the following:
```
Letting `k := labels[i]` and `P := pred_probs[i]`  denote the given label and predicted class-probabilities for the `i`th datapoint, its score can either be: 
'normalized_margin' := P[k] - max_{k' != k}[ P[k'] ] 
'self_confidence' := P[k]
'confidence_weighted_entropy' := entropy(P) / self_confidence
```

Reply:
Sounds good! I made these changes. Thanks.

35624d6f3e43276c53967e531e123b123f3013a7